ci/unit_tests/functions_deploy/main_test.py

```diff
@@ -0,0 +1,198 @@
+"""
+Copyright 2019 IBM Corporation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json, os, requests, shutil, unittest, urllib, uuid, zipfile
+
+import functions_deploy
+from ...test_utils import BaseTestCaseCapture
+
+class TestMain(BaseTestCaseCapture):
+
+    dataBasePath = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'main_data')
+    packageBase = "Package-for-WAW-CI-"
+
+    def setup_class(cls):
+        BaseTestCaseCapture.checkEnvironmentVariables(['CLOUD_FUNCTIONS_USERNAME', 'CLOUD_FUNCTIONS_PASSWORD'])
+        cls.cloudFunctionsUrl = os.environ.get('CLOUD_FUNCTIONS_URL',
+                                               'https://us-south.functions.cloud.ibm.com/api/v1/namespaces')
+        cls.namespace = os.environ.get('CLOUD_FUNCTIONS_NAMESPACE', "Prague Cognitive Services_IoT-Prague")
+        cls.urlNamespace = urllib.quote(cls.namespace)
+
+    def callfunc(self, *args, **kwargs):
+        functions_deploy.main(*args, **kwargs)
+
+    def _getFunctionsInPackage(self, package):
+        functionListUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/?limit=0&skip=0'
+
+        functionListResp = requests.get(functionListUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
```

Comment:
We could probably set CLOUD_FUNCTIONS_USERNAME and CLOUD_FUNCTIONS_PASSWORD similarly as CLOUD_FUNCTIONS_URL in setup_class to have it same.

Reply:
changed